test_util/test_upgrade_vpc.py

```diff
@@ -43,6 +45,64 @@
 log = logging.getLogger(__name__)
 
 
+def _parse_task_details(apps_tasks_details):
+    task_idx = 0    # We have a single app and a task for this test.
+
+    try:
+        apps_tasks_json = json.loads(apps_tasks_details)
+    except (TypeError, json.JSONDecodeError):
+        logging.debug("Failed to parse {apps_tasks_details}".format(apps_tasks_details=apps_tasks_details))
+        raise
+
+    try:
+        tasks_state = apps_tasks_json["tasks_details"]["tasks"][task_idx]["state"]
+        health_check_interval = \
+            apps_tasks_json["apps_details"]["apps"][task_idx]["healthChecks"][task_idx]["intervalSeconds"]
+        task_idx = apps_tasks_json["tasks_details"]["tasks"][task_idx]["id"]
+        last_success = apps_tasks_json["tasks_details"]["tasks"][task_idx]["healthCheckResults"][task_idx]["lastSuccess"]
+    except IndexError as e:
+        logging.debug("Failed to get task detail: {exp}".format(exp=str(e)))
+        return None
+
+    TaskDetails = collections.namedtuple("TaskDetails", "state id health_check_interval last_success_time")
+
+    task_details = TaskDetails(
+        state=tasks_state,
+        id=task_idx,
+        health_check_interval=datetime.timedelta(seconds=health_check_interval),
+        last_success_time=(datetime.datetime.strptime(last_success, "%Y-%m-%dT%H:%M:%S.%fZ")))
+
+    return task_details
+
+
+def _signal_failure(result=-1):
+    log.info("Test failed! VPC will remain for debugging 1 hour from instantiation")
+    sys.exit(result)
```

Comment:
Our CI relies on nullification of error codes to mute tests: https://github.com/dcos/dcos/blob/master/test_util/test_aws_cf.py#L124

Try to follow the same testing pattern where we only sys.exit() at the very end with a code that we can intervene if necessary.